### ISSUE TITLE
Update docs related to inline configuration comment descriptions

### ISSUE THREAD

**Issue by DesselBane** [eslint/eslint#20528]
### Rule details

Flag any use of an inline configuration which has no description ( https://eslint.org/docs/latest/use/configure/rules#use-configuration-comments )

### Related ECMAScript feature

Nor an ECMA feature but I think this still makes more sense as a rule, though it could also work like `reportUnusedDisableDirectives`

### What type of rule is this?

Warns about a potential problem

### Example code

```js
Warns about a disabled rule if disabled using an inline comment. Typically this means that there is some special reason why the rule is disabled for this one line only and this reason must be written down.

Error:

// eslint-disable-next-line unicorn/no-immediate-mutation
isMobileIntegration.value = false;


Ok

// eslint-disable-next-line unicorn/no-immediate-mutation -- Intentional to trigger setter
isMobileIntegration.value = false;
```

### Why should this rule be in the core instead of a plugin?

How comments work is something specific to eslint itself so I would have the rule in eslint itself.

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by lumirlumir** [comment]
Hi @DesselBane, thanks for the issue.

I believe this is already implemented and maintained by the ESLint Community’s `eslint-plugin-eslint-comments` rule, [`require-description`](https://eslint-community.github.io/eslint-plugin-eslint-comments/rules/require-description.html). Would that fit your use case?

**Comment by DesselBane** [comment]
Awesome this is exactly what I was looking for, thanks @lumirlumir and sorry for the spam. 

Would it help if we add a comment with a link to this plugin in the docs? If yes I'd be willing to contribute a PR but only if you think it makes sense.

**Comment by lumirlumir** [comment]
I think it's a good idea to add documentation. 👍

But before moving forward, I'd like to hear one more opinion from the team. @eslint/eslint-team

**Comment by snitin315** [comment]
We can definitely mention this in the docs. @DesselBane Feel free to raise a PR.
